service/src/main/kotlin/fi/espoo/evaka/pis/EmployeeQueries.kt

```diff
@@ -298,3 +302,24 @@ fun Database.Read.isPinLocked(employeeId: UUID): Boolean =
         .bind("id", employeeId)
         .mapTo<Boolean>()
         .firstOrNull() ?: false
+
+/*
+Last login timestamps are tracked since May 2021 so using June 2021 as a default should make sure that this role reset
+is not too eager
+*/
+fun Database.Transaction.clearRolesForInactiveEmployees(now: Instant) = createUpdate(
+    """
+WITH reset_employee AS (
+    UPDATE employee
+    SET roles = '{}'
+    WHERE coalesce(last_login, :defaultLastLogin) < :now::timestamp with time zone - interval '3 months'
+    RETURNING id
+), deleted_daycare_acl AS (
+    DELETE FROM daycare_acl WHERE employee_id IN (SELECT id FROM reset_employee)
+)
+DELETE FROM daycare_group_acl WHERE employee_id IN (SELECT id FROM reset_employee)
+"""
+)
+    .bind("now", now)
+    .bind("defaultLastLogin", ZonedDateTime.of(LocalDateTime.of(2021, 6, 1, 0, 0), helsinkiZone).toInstant())
```

Comment:
Would `HelsinkiDateTime.of(LocalDate.of(2021, 6, 1), LocalTime.of(0, 0))` be simpler? No need to manually fiddle with zones or cast to Instant, since it works with JDBI correctly

Reply:
Yeah, that's a good idea. I've never used it so it didn't even cross my mind.